### PR TITLE
check_trigger_amount() will now output the changes 

### PR DESCRIPTION
**Pull request by vinferrer**
Closes #314 

#314 established that inputs to `BlueprintInput.check_trigger_ammount`` should remain unmodified outside of the class (i.e., any changes to its inputs should be done on copies, rather than the original objects). This PR solves that problem by returning the outputs so they can be modified outside the class if desired.

## Proposed Changes

  -  `check_trigger_amounts`  no longer modifies inputs
  -  Now the results are returned so they can be modified outside the method
  -

## Change Type

- [ ] `bugfix` (+0.0.1)
- [ ] `minor` (+0.1.0)
- [ ] `major`  (+1.0.0)
- [x] `refactoring` (no version update)
- [ ] `test` (no version update)
- [ ] `infrastructure` (no version update)
- [ ] `documentation` (no version update)
- [ ] `other`

## Checklist before review

- [ ] I added everything I wanted to add to this PR.
- [ ] \[Code or tests only\] I wrote/updated the necessary docstrings.
- [ ] \[Code or tests only\] I ran and passed tests locally.
- [ ] \[Documentation only\] I built the docs locally.
- [ ] My contribution is harmonious with the rest of the code: I'm not introducing repetitions.
- [ ] My code respects the adopted style, especially linting conventions.
- [ ] The title of this PR is explanatory on its own, enough to be understood as part of a changelog.
- [ ] I added or indicated the right labels.

- [ ] I added information regarding the timeline of completion for this PR.

- [ ] Please, comment on my PR while it's a draft and give me feedback on the development!

### CODE REVIEW

**Comment by eurunuela** [comment]
@vinferrer if the PR is ready for review, please make it ready. It's currently a draft.

**Comment by vinferrer** [comment]
i wanted to try the tests and wait for #349

**Comment by smoia** [comment]
Hello @vinferrer!
Sorry for letting this slip out of my sight until now.

I'm afraid this is _not_ what #314 is about. The idea there is not to output the computations of `check_trigger_amount()`, but rather to be **sure** that what is provided as input to our objects (`BlueprintInput` and `BlueprintOutput`) stays unmodified after something operates on the objects.

The issue, at the moment, is that our objects use a shallow copy of their input, and when we call internal methods (like `check_trigger_amount()`) or we apply a resampling, the original input gets changed too.

If you want to close #314, you need to modify [line 92](https://github.com/physiopy/phys2bids/blob/master/phys2bids/tests/test_physio_obj.py#L92), [line 104](https://github.com/physiopy/phys2bids/blob/master/phys2bids/tests/test_physio_obj.py#L104), and [line 143](https://github.com/physiopy/phys2bids/blob/master/phys2bids/tests/test_physio_obj.py#L143) of `test_physio_obj.py` by removing `.copy()` methods and see if you're passing the tests or not.

I'm afraid I don't see the reason for making `check_trigger_amount()` return general outputs rather than changing directly object attributes, especially since you are still doing so anyway. Can you ()or anyone else that is in favour of the change) explain a bit further why is this a good idea?

**Comment by smoia** [comment]
Well, you did prove me wrong - it's passing tests as it should. @tsalo, is this the implementation you had in mind?

**Comment by vinferrer** [comment]
> Well, you did prove me wrong - it's passing tests as it should. @tsalo, is this the implementation you had in mind?

This made the PR worth it.

File: phys2bids/physio_obj.py [hunk 1]
```diff
@@ -439,16 +439,16 @@ def check_trigger_amount(self, thr=None, num_timepoints_expected=0, tr=0):
...
+        :
+        thr: float
             Threshold used by the function to detect trigger points.
-        self.num_timepoints_found: int
+        num_timepoints_found: int
             Property of the `BlueprintInput` class.
             Contains the number of timepoints found
             with the automatic estimation.
-        self.timeseries:
+        shited_timeseries:
```
Comment (eurunuela) [inline phys2bids/physio_obj.py:451]: Eeh... were you angry when writing this @vinferrer ? 😅

**Review by eurunuela — COMMENTED** [review]
I have to say I'm not a fan of this solution either. I'd say that the whole point of using classes is that you get to directly work on the variables without needing to return them.

If the point is to avoid original time series getting overwritten, I'd keep copies of the originals in `__init__` as @tsalo proposed.

File: phys2bids/physio_obj.py (hunk 1 above)
Comment (vinferrer) [reply phys2bids/physio_obj.py:451]: Lol, just tired

**Comment by vinferrer** [comment]
> If you want to close #314, you need to modify [line 92](https://github.com/physiopy/phys2bids/blob/master/phys2bids/tests/test_physio_obj.py#L92), [line 104](https://github.com/physiopy/phys2bids/blob/master/phys2bids/tests/test_physio_obj.py#L104), and [line 143](https://github.com/physiopy/phys2bids/blob/master/phys2bids/tests/test_physio_obj.py#L143) of `test_physio_obj.py` by removing `.copy()` methods and see if you're passing the tests or not.

Hi people I just opened #357 to see if this the tests passed by modifying these lines and they pass. The test also pass in my computer both linux and windows OS. I believe this behavior is limited to stefano's computer for the moment.

**Comment by eurunuela** [comment]
> I believe this behavior is limited to stefano's computer for the moment.

What do you mean by "this behavior"? @tsalo is the one who opened the issue.

**Comment by vinferrer** [comment]
> > I believe this behavior is limited to stefano's computer for the moment.
> 
> What do you mean by "this behavior"? @tsalo is the one who opened the issue.

@tsalo opened the issue, but the bug was reported by stefano as you can see in #314

**Comment by vinferrer** [comment]
I am gonna close this PR for the moment, lets recover the conversation in #314

**Pull request closed without merging**